### PR TITLE
Extract Babel presets

### LINKED ISSUE #947: Option to disable babel-plugin-transform-runtime?
I am porting a Node.js v6 project to use AVA. I can already specify just the Babel plugins I need for my tests to run:

``` json
  "ava": {
    "babel": {
      "plugins": [
        "transform-async-to-generator",
        "transform-es2015-modules-commonjs"
      ]
    },
  }
```

However AVA will always add the `babel-plugin-transform-runtime` plugin. It's not really needed for my use case and actually causes a test to break:

I'm testing whether a module returns a promise, using `t.true(thingThatReturnsAPromise() instanceof Promise)`. The `babel-plugin-transform-runtime` plugin replaces `Promise` within the test with a `core-js` implementation, but the module returns a native promise. Hence `instanceof` fails.

It'd be useful if the `babel-plugin-transform-runtime` could be disabled. Perhaps like this:

``` json
"ava": {
  "babel": {
    "disable-transform-runtime": true
```

The above would leave AVA's Babel config in default mode, but disable the plugin.

To use it with `inherit` mode:

``` json
"ava": {
  "babel": {
    "inherit": true,
    "disable-transform-runtime": true
```

Or to use it with a custom config:

``` json
"ava": {
  "babel": {
    "disable-transform-runtime": true,
    "plugins": [
      "transform-async-to-generator",
      "transform-es2015-modules-commonjs"
    ]
```

Thoughts?

### PR DESCRIPTION
**Pull request by novemberborn**
Removes deprecated assertions (breaking!)

Uses the new Babel presets (#1193, #1195). Will publish those dependencies and update the `package.json` references when CI passes.

Since this removes `babel-runtime` we can close #947 and #1089.

### CODE REVIEW

File: lib/babel-config.js [hunk 1]
```diff
@@ -25,68 +24,12 @@ function validate(conf) {
...
-		}
-
-		return value;
+function makeTransformTestFiles(powerAssert) {
+	return babel => {
+		return require('@ava/babel-preset-transform-test-files')(babel, {powerAssert});
 	};
 }
 
-const defaultPresets = lazy(() => {
```
Comment (sindresorhus) [inline lib/babel-config.js]: @novemberborn Do you remember why we have laziness here? There must be a reason. Maybe we should preserve the laziness?

Comment (novemberborn) [inline lib/babel-config.js]: It seemed unnecessary yesterday but I think I figured it out. This config is used even if all transforms have been cached. So we should keep the laziness in place.

Comment (novemberborn) [inline lib/babel-config.js]: Restored the lazy behavior, though in a somewhat simplified form.

**Comment by tylerjharden** [comment]
So this is good to go? And fixes "warning ava > babel-preset-es2015-node4@2.1.1: Use npmjs.org/babel-preset-env instead, see https://github.com/babel/babel-preset-env"?

Very exciting if so.

**Comment by sindresorhus** [comment]
@novemberborn I tried this PR on some of my projects, like Got, Pageres, etc, and seems to be working well.

**Comment by sindresorhus** [comment]
@tylerjharden Yes, it removes the warning. It's good to go when it's done.

**Comment by novemberborn** [comment]
> It's good to go when it's done.

@sindresorhus Could you approve it then? ;-) CI failure is just AppVeyor being confused.

**Comment by sindresorhus** [comment]
@novemberborn Don't you need to publish the extracted modules first and update package.json? Right now they reference Git commits.

**Comment by novemberborn** [comment]
> Don't you need to publish the extracted modules first and update package.json? Right now they reference Git commits.

Hence the `DO NOT MERGE` label 😉

**Comment by sindresorhus** [comment]
It's approved. :shipit: :shipit: :shipit: :shipit:

**Comment by sindresorhus** [comment]
1ab7e29c-117e-11e6-9cbb-f0828bacea8e

**Comment by mightyiam** [comment]
Can't wait for next release.

**Comment by danez** [comment]
This PR removed the stage-2 preset and broke my test when updating as I was using object-spread. Should I change the README and remove the mention of stage-2 or was that removed by mistake?

**Comment by novemberborn** [comment]
@danez yes, this sentence shouldn't mention `stage-2`:

> We use our own bundled Babel with our @ava/stage-4 and stage-2 preset, as well as custom transforms for test and helper files.

We would appreciate a PR!

**Pull request merged**